lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1206 @@
+
+    note = None
+
+    if record['able_to_test'] == 'no':
+        note = 'never-tested'
+    else:
+        note = 'can-test'
+
+    specimen_type = 'NSECR'  # Nasal swab.
+    specimen_resource = create_specimen_resource(
+        [specimen_identifier], patient_reference, specimen_type, received_time,
+        collection_date(record, collection_method), note
+    )
+
+    return create_entry_and_reference(specimen_resource, "Specimen")
+
+
+def collection_date(record: dict, collection_method: CollectionMethod) -> Optional[str]:
+    """
+    Determine sample/specimen collection date from the given REDCap *record*.
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        return record["nasal_swab_q"]
+    elif collection_method == CollectionMethod.SWAB_AND_SEND:
+        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
+    else:
+        return None
+
+def combine_multiple_fields(record: Dict[Any, Any], field_prefix: str, field_suffix: str = "") -> Optional[List]:
+        """
+        Handles the combining of multiple fields asking the same question such
+        as country and state traveled.
+        """
+        regex = rf'^{re.escape(field_prefix)}[0-9]+{re.escape(field_suffix)}$'
+        empty_value = ''
+        answered_fields = list(filter(lambda f: filter_fields(f, record[f], regex, empty_value), record))
+
+        if not answered_fields:
+            return None
+
+        return list(map(lambda x: record[x], answered_fields))
+
+
+
+def create_enrollment_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    # Do not include `core_age_years` because we calculate the age ourselves in the computed questionnaire.
+    integer_questions = [
+        'weight',
+        'height_total'
+    ]
+
+    string_questions = [
+        'study_area',
+        'text_or_email',
+        'text_or_email_attestation',
+        'campus_location',
+        'affiliation',
+        'student_level',
+        'sea_employee_type',
+        'core_house_members',
+        'core_education',
+        'core_income',
+        'vaccine_method',
+        'wfh_base',
+        'core_housing_type',
+        'core_health_risk',
+        'core_tobacco_use',
+        'sought_care_base',
+        'hospital_where_base',
+        'hospital_ed_base',
+        'prior_test_positive_base',
+        'prior_test_type_base',
+        'prior_test_result_base',
+        'contact_base',
+        'wash_hands_base',
+        'clean_surfaces_base',
+        'hide_cough_base',
+        'mask_base',
+        'distance_base',
+        'bmi',
+        'novax_reason',
+        'covid_vaccine',
+        'vaccine_where',
```

Comment:
This is being ingested as "3" (e.g. record 61) instead of the expected string value

Reply:
In REDCap the options for vaccine_where are:
1, Medical clinic or hospital
2, Pharmacy or drugstore
3, Workplace
4, School
5, Other

I changed the ETL so that this gets brought in as an integer. (It was a string.)

Reply:
No, we want it to be ingested as a string. This is a REDCap side thing that needs to be fixed.

Reply:
This variable was in the list whose values Misja fixed from integer to string in REDCap today. I'm ingesting this as a string.